openedx/core/djangoapps/schedules/migrations/0009_schedule_copy_column_values.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.26 on 2019-11-24 20:52
+from __future__ import unicode_literals
+
+from django.db import migrations, transaction
+
+
+def copy_column_value_forwards(apps, schema_editor):
+    """
+    Copy the start field into start_date field
+
+    This table has around 25 million rows, we'll follow non-atomic migrations.
+    https://docs.djangoproject.com/en/2.2/howto/writing-migrations/#non-atomic-migrations
+    """
+    Schedule = apps.get_model('schedules', 'Schedule')
+    while Schedule.objects.filter(start_date__isnull=True).exists():
+        with transaction.atomic():
+            for row in Schedule.objects.filter(start_date__isnull=True)[:1000]:
+                row.start_date = row.start
+                row.save()
```

Comment:
Are you certain that there's no `NULL` values in the existing `start` fields? If so, seems like this data migration would be an infinite loop.

Reply:
The current start field is a non-null field (null=False).  I would still check the database to be sure.

Reply:
<img width="462" alt="image" src="https://user-images.githubusercontent.com/17911020/70174324-7b9eec00-16f6-11ea-9cdf-a0e0bde0bd20.png">

Reply:
@doctoryes There are no Null values in ``start``. There's also an example here
https://docs.djangoproject.com/en/2.2/howto/writing-migrations/#non-atomic-migrations
I don't see why it would be an infinite loop.

Reply:
OK - this point seems fine then.